### PR TITLE
LPS-90074 GraphQL doesn't support generic type set on Page

### PR DESCRIPTION
**Pull request by inacionery**
To get the schema

curl 'http://localhost:8080/o/headless-workflow-graphql/schema.json' 

Sample of query 

curl 'http://localhost:8080/o/headless-workflow-graphql' -H 'Content-Type: application/json' -H 'Accept: application/json' --data-binary '{"query":"query {\n  getWorkflowLog(id: 1){\n    auditPerson\n    commentLog\n    dateCreated\n    id\n    person\n    previousPerson\n    previousState\n    self\n    state\n    type\n  }\n  getWorkflowTask ( id: 1 ){\n    completed\n    dateCompleted\n    dateCreated\n    definitionName\n    description\n    dueDate\n    id\n    name\n    self\n    transitions\n  }\n}","variables":{}}'

/cc @brianchandotcom

### CODE REVIEW

**Comment by brianchandotcom** [comment]
Awesome, thank you!

**Comment by Preston-Crary** [comment]
@inacionery how do you make a mutation call?

**Comment by inacionery** [comment]
Hey @Preston-Crary

Sorry for not having informed an example of mutation before, this should work:

```
 curl http://localhost:8080/o/headless-workflow-graphql' \
  -H 'Content-Type: application/json' \
  -H 'Accept: application/json' \
  -d '{
    "query": "mutation($id:Int) { postWorkflowTasksAssignToMe(input:$id) { id  } }",
    "variables": {
      "id": 1
    }
  }'
```

**Comment by Preston-Crary** [comment]
@inacionery Thanks!

**Comment by shuyangzhou** [comment]
@inacionery with the mutation example, I get 
```
2019-02-07 01:07:09.715 WARN  [http-nio-8080-exec-1][GraphQL:515] Query failed to validate : 'mutation($id:Int) { postWorkflowTasksAssignToMe(input:$id) { id  } }'
```
Nothing is executed, can you double check?

**Comment by inacionery** [comment]
Hey @shuyangzhou @Preston-Crary 

I'm sorry, I don't replace the variable in the query, follows the right query

  curl 'http://localhost:8080/o/headless-workflow-graphql' \
  -H 'Content-Type: application/json' \
  -H 'Accept: application/json' \
  -d '{
    "query": "mutation($id:Long) { postWorkflowTasksAssignToMe(id:$id) { id  } }",
    "variables": {
      "id": 1
    }
  }'

**Comment by Preston-Crary** [comment]
Hi @inacionery, can you rebase this?

**Comment by inacionery** [comment]
Sure!

File: modules/apps/headless/headless-workflow/headless-workflow-impl/src/main/java/com/liferay/headless/workflow/internal/resource/v1_0/BaseObjectResourceImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	protected <T, R> List<R> transform(
+		List<T> list, Function<T, R> transformFunction) {
+
+		return TransformUtil.transform(list, transformFunction);
+	}
+
+	@Context
+	protected AcceptLanguage acceptLanguage;
+
+	@Context
```
Comment (Preston-Crary) [inline modules/apps/headless/headless-workflow/headless-workflow-impl/src/main/java/com/liferay/headless/workflow/internal/resource/v1_0/BaseObjectResourceImpl.java]: Hi @inacionery,

can you show us an example using one of the context fields?

Not sure if there is a way to defined these in the schema.

Thanks again

Comment (inacionery) [reply modules/apps/headless/headless-workflow/headless-workflow-impl/src/main/java/com/liferay/headless/workflow/internal/resource/v1_0/BaseObjectResourceImpl.java]: Ok, I'll start working on it now

**Comment by inacionery** [comment]
Hey Preston,

you need to activate the workflow for an asset and submit it. Then you need to pass the RoleId to the query.

`curl 'http://localhost:8080/o/headless-workflow-graphql' -H 'Accept-Encoding: gzip, deflate, br' -H 'Content-Type: application/json' -H 'Accept: application/json' --data-binary '{"query":"\nquery ($pagination: InputPagination!){\n  getRolesWorkflowTasksPage(roles_45_id: 20105, Pagination: $pagination){\n    completed\n    dateCompleted\n    dateCreated\n    definitionName\n    description\n    dueDate\n    id\n    name\n  }\n}\n\n","variables":{"pagination":{"itemsPerPage":10,"pageNumber":1}}}' --compressed`

**Comment by Preston-Crary** [comment]
Please see https://github.com/shuyangzhou/liferay-portal/pull/7366

**Pull request closed without merging**